### ISSUE TITLE
deleting experiments

### ISSUE THREAD

**Issue by remcinerney** [nextml/NEXT#177]
Hi guys,
Is it possible to delete experiments once they have been launched? I can't find this in the documentation...

**Comment by remcinerney** [comment]
OK great, I'll use the helper class... thanks.

**Comment by stsievert** [comment]
You can also remove your experiments with `docker-compose rm`. I recommend this method.

If you're launching via AMI, launching a new machine via the AMI will (effectively) remove all experiments. You can restore any experiments you wish to keep by downloading/restoring as mentioned at the bottom of the experiment list.

Not providing an interface to remove data is a design goal of NEXT.

**Comment by remcinerney** [comment]
Is there a convenient way to ssh into an AWS cluster using the next_ec2 script? (I'm trying to do this so I can play around with the docker containers as discussed in this thread....

Thanks

**Comment by lalitkumarj** [comment]
Yes.

python next_ec2.py --identity-file={location to pem key} --key-pair={name
of pem key} docker_login MACHINE_NAME

You should also have the appropriate environment keys set:

AWS_SECRET_ACCESS_KEY
AWS_ACCESS_KEY_ID
AWS_BUCKET_NAME

See the following for more detail:

https://github.com/nextml/NEXT/blob/master/README.md

Note that to use this above procedure, the machine should have been
launched by the next_ec2 script as well as described on that page. So
basically using a python next_ec2.py launch command.

You can also SSH into the AMI using a PEM key. Make sure that you become
superuser (su -) and then go to the /usr/local/next-discovery directory.

I hope that helps!

**Comment by remcinerney** [comment]
Ok great - I now have access to through that method - but not sure which container I should be removing to delete experiments?
Is each experiment operating in a separate container?

**Comment by lalitkumarj** [comment]
The idea is that you delete all the containers which deletes ALL the
experiments. Then you do docker-compose up to fire up NEXT again. There is
no way to really delete one experiment - this was done on purpose to ensure
that experiments don't get deleted by mistake.

Lalit

**Comment by remcinerney** [comment]
Ok got it, makes sense

In that case is there a difference between using docker and docker-compose when doing this?

**Comment by lalitkumarj** [comment]
Yes. You should always use docker-compose (unless bad things happen and you
need to do a reset).

docker-compose stop
docker-compose start
docker-compose up
docker-compose rm
docker-compose log

are all the commands you probably need.
